### PR TITLE
Add CODEOWNERS

### PR DESCRIPTION
**Pull request by annaswims**
## Description of change
After a repository admin has made the change, CODEOWNERS will require a review from someone on the `@department-of-veterans-affairs/backend-review-group` `@department-of-veterans-affairs/va-api-engineers`  or `@department-of-veterans-affairs/ad-hoc-devops` in order to merge a PR.
## Testing done
none

## Acceptance Criteria (Definition of Done)
- [ ] Change has been communicated to team.
- [ ] - Admin has [required Codeowners ](https://help.github.com/assets/images/help/repository/PR-review-required-code-owner.png)

### CODE REVIEW

**Comment by annaswims** [comment]
I believe that this PR can be merged in advance of the admin work. It just won't take effect until the admin changes the setting.

**Comment by annaswims** [comment]
I audited the last month of PR's and this set of rules will allow for everyone to continue working as usual, just with more formalized rules.

The only other change nessary is it looks like @rtravitz should be added to @department-of-veterans-affairs/va-api-engineers

**Comment by rtravitz** [comment]
> The only other change nessary is it looks like @rtravitz should be added to @department-of-veterans-affairs/va-api-engineers

Oh, thanks for noticing that @annaswims! I actually see a few more VA API engineers missing from that group who should probably be in there. I can bring it up with them!

**Comment by dhall** [comment]
Hi @annaswims! Are you looking to have all the engineers that work on vets-api in this group? I can get a list of those on VA Lighthouse that do so, if that would be helpful.

**Comment by annaswims** [comment]
Hi, @dhall! Now that VA.gov project has been spit up into multiple contracts, we want to enforce that each app team must have a PR review from the platform team and Lighthouse developers will continue to have a PR review from someone on the `@department-of-veterans-affairs/va-api-engineers` team.  Not all developers who have write access to vets-api will be in on a CODEOWNERS team though because app teams will still require a review from the platform team.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# These owners will be the default owners for everything in
+# the repo. Unless a later match takes precedence,
+#  @department-of-veterans-affairs/backend-review-group will be requested for
+# review when someone opens a pull request.
+
+*       @department-of-veterans-affairs/backend-review-group @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/ad-hoc-devops
+
+*.md    @department-of-veterans-affairs/teams/ad-hoc-general @department-of-veterans-affairs/backend-review-group @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/ad-hoc-devops
```
Comment (kfrz) [inline .github/CODEOWNERS]: @annaswims do you have background on why `*.md` has a different precedence order?

Comment (annaswims) [reply .github/CODEOWNERS]: I audited PR's from the last month and there was one PR that was documentation only that was approved by a product person. But on second thought it's not that hard to get a 👍 from a dev in one of these groups so I think I'll just take out this line.

**Comment by annaswims** [comment]
I'm going to hold off on this for a bit.  I learned that github tags all codeowner teams, even if the PR author tags the single relevant team.  Looking to see if there are ways to make it less distracting.

**Comment by annaswims** [comment]
maybe making `@department-of-veterans-affairs/va-api-engineers` the owners of 
```
modules/appeals_api
modules/claims_api
modules/openid_auth
modules/va_facilities
modules/vba_documents
modules/veteran
modules/veteran_verification 
```

and `@department-of-veterans-affairs/backend-review-group` the owners of everything else won't be too tedious?

**Comment by annaswims** [comment]
@dvogel As the “Platform” part of VSP ramps up, we’re feeling the need to put stronger protections on who can merge code. As we onboard more app teams, especially those that aren’t VSA/VSP we’d like to require a review from a member of either department-of-veterans-affairs/teams/va-api-engineers or department-of-veterans-affairs/teams/backend-review-group.  In practice, It’s not any different that what we’re doing today, other than that it will be enforced by GitHub.  
However, due to an unfortunate choice by GitHub, each group will automatically be tagged for review when a PR is created.  It’ll be up to the PR author to untag the team(s) that they don’t want a review from at that time. I know this will be noisy, but I hope it’s something your team can put up with.

**Comment by dvogel** [comment]
> @dvogel As the “Platform” part of VSP ramps up, we’re feeling the need to put stronger protections on who can merge code. As we onboard more app teams, especially those that aren’t VSA/VSP we’d like to require a review from a member of either department-of-veterans-affairs/teams/va-api-engineers or department-of-veterans-affairs/teams/backend-review-group. In practice, It’s not any different that what we’re doing today, other than that it will be enforced by GitHub.
> However, due to an unfortunate choice by GitHub, each group will automatically be tagged for review when a PR is created. It’ll be up to the PR author to untag the team(s) that they don’t want a review from at that time. I know this will be noisy, but I hope it’s something your team can put up with.

Thanks for the heads-up @annaswims. We'll do our best to determine which PRs are well-positioned to contribute to and which teams should be on PRs we make.

**Pull request merged**